great_expectations/zep/postgres_datasource.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import Dict, List, Optional, Set
+
+from great_expectations.core.batch_spec import SqlAlchemyDatasourceBatchSpec
+from great_expectations.execution_engine import SqlAlchemyExecutionEngine
+from great_expectations.zep.interfaces import (
+    Batch,
+    BatchRequest,
+    BatchRequestOptions,
+    DataAsset,
+    Datasource,
+)
+
+
+class PostgresDatasourceError(Exception):
+    pass
+
+
+@dataclasses.dataclass(frozen=True)
+class ColumnSplitter:
+    method_name: str
+    column_name: str
+    namespace: str
+    template_params: List[str]
+
+
+class TableAsset(DataAsset):
+    def __init__(
+        self,
+        datasource: Datasource,
+        name: str,
+        table_name: str,
+        # TODO(bdirks): Provide a mechanism to add splitters in __init__
```

Comment:
Is there more to it than accepting a `column_splitter: Optional[ColumnSplitter] = None` keyword?

or is the difficulty dealing with multiple column splitters (🤷 ) and the logic in `get_batch_list_from_batch_request()`?